tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/LiferayGradleProject.java

```diff
@@ -354,4 +360,84 @@ private IFolder _createResorcesFolder(IProject project) {
 
 	private static final String[] _ignorePaths = {".gradle", "build", "dist", "liferay-theme.json"};
 
+	public void watch() {
+		Job[] jobs = Job.getJobManager().find(_watch_job_name);
+
+		if (ListUtil.isNotEmpty(jobs)) {
+			return;
+		}
+
+		Job job = new Job(_watch_job_name) {
+
+			public boolean belongsTo(Object family) {
+				return _watch_job_name.equals(family);
+			}
+
+			@Override
+			public IStatus run(IProgressMonitor monitor) {
+				IStatus status = Status.OK_STATUS;
+
+				try {
+					GradleUtil.runGradleTask(getProject(), new String[] {
+						"watch"
+					}, new String[] {
+						"-t"
+					}, monitor);
+				}
+				catch (CoreException e) {
+					status = GradleCore.createErrorStatus(e);
+				}
+
+				return status;
+			}
+
+		};
+
+		job.setSystem(true);
+
+		job.schedule();
+	}
+
+
+	@Override
+	public void unwatch() {
+		Job[] jobs = Job.getJobManager().find(_watch_job_name);
+
+		if (ListUtil.isNotEmpty(jobs)) {
+			jobs[0].cancel();
+		}
+	}
+
+	@Override
+	public boolean supported() {
+		boolean supported = false;
+
+		ProjectConnection connection = null;
```

Comment:
This is really slow.  Can we figure out some other way to determine if watch is supported?